node/pkg/watchers/solana/shim_test.go

```diff
@@ -1007,3 +1003,1284 @@ func TestShimFromIntegratorWithMultipleShimTransactions(t *testing.T) {
+				"Program EtZMZM22ViKMo4r5y4Anovs3wKQ2owUmDpjygnMMcdEX success"
+			],
+			"postBalances": [
+				499999999997496260, 1057920, 2350640170, 1552080, 946560, 1, 1141440, 0,
+				1169280, 1009200, 1141440
+			],
+			"postTokenBalances": [],
+			"preBalances": [
+				500000000000000000, 1057920, 2350640070, 0, 0, 1, 1141440, 0, 1169280,
+				1009200, 1141440
+			],
+			"preTokenBalances": [],
+			"rewards": [],
+			"status": {
+				"Ok": null
+			}
+		},
+		"slot": 3,
+		"transaction": {
+			"message": {
+				"header": {
+					"numReadonlySignedAccounts": 0,
+					"numReadonlyUnsignedAccounts": 6,
+					"numRequiredSignatures": 1
+				},
+				"accountKeys": [
+					"H3kCPjpQDT4hgwWHr9E9pC99rZT2yHAwiwSwku6Bne9",
+					"2yVjuQwpsvdsrywzsJJVs9Ueh4zayyo5DYJbBNc3DDpn",
+					"9bFNrXNb2WTx8fMHXCheaZqkLZ3YCCaiqTftHxeintHy",
+					"9vohBn118ZEctRmuTRvoUZg1B1HGfSH8C5QX6twtUFrJ",
+					"HeccUHmoyMi5S6nuTcyUBh4w4me3FP541a52ErYJRT8a",
+					"11111111111111111111111111111111",
+					"EtZMZM22ViKMo4r5y4Anovs3wKQ2owUmDpjygnMMcdEX",
+					"HQS31aApX3DDkuXgSpV9XyDUNtFgQ31pUn5BNWHG2PSp",
+					"SysvarC1ock11111111111111111111111111111111",
+					"SysvarRent111111111111111111111111111111111",
+					"worm2ZoG2kUd4vFXhvjh93UUH596ayRfgQ2MgjNMTth"
+				],
+				"recentBlockhash": "CMqPGm4icRdNuHsWJUK4Kgu4Cbe2nDQkYNqugQkKPa4Y",
+				"instructions": [
+				],
+				"indexToProgramIds": {}
+			},
+			"signatures": [
+				"3NACxoZLehbdKGjTWZKTTXJPuovyqAih1AD1BrkYj8nzDAtjiQUEaNmhkoU1jcFfoPTAjrvnaLFgTafNWr3fBrdB"
+			]
+		},
+		"version": "legacy"
+	}
+	`
+
+	///////// A bunch of checks to verify we parsed the JSON correctly.
+	var txRpc rpc.TransactionWithMeta
+	err := json.Unmarshal([]byte(eventJson), &txRpc)
+	require.NoError(t, err)
+
+	tx, err := txRpc.GetParsedTransaction()
+	require.NoError(t, err)
+
+	require.Equal(t, 0, len(tx.Message.Instructions))
+	require.Equal(t, 1, len(txRpc.Meta.InnerInstructions))
+
+	///////// Now we start the real test.
+
+	logger := zap.NewNop()
+	msgC := make(chan *common.MessagePublication, 10)
+	s := shimNewWatcherForTest(t, msgC)
+	require.True(t, s.shimEnabled)
+
+	var whProgramIndex uint16
+	var shimProgramIndex uint16
+	var shimFound bool
+	for n, key := range tx.Message.AccountKeys {
+		if key.Equals(s.contract) {
+			whProgramIndex = uint16(n)
+		}
+		if key.Equals(s.shimContractAddr) {
+			shimProgramIndex = uint16(n)
+			shimFound = true
+		}
+	}
+
+	require.Equal(t, uint16(10), whProgramIndex)
+	require.True(t, shimFound)
+	require.Equal(t, uint16(6), shimProgramIndex)
+
+	alreadyProcessed := ShimAlreadyProcessed{}
+	found, err := s.shimProcessTopLevelInstruction(logger, whProgramIndex, shimProgramIndex, tx, txRpc.Meta.InnerInstructions, 1, alreadyProcessed, false)
+	require.ErrorContains(t, err, "topLevelIndex 1 is greater than 0")
+	require.False(t, found)
+	require.Equal(t, 0, len(s.msgC))
+	require.Equal(t, 0, len(alreadyProcessed))
+}
+
+func TestShimInnerShouldFailIfNoInstructionsShouldFail(t *testing.T) {
```

Comment:
```suggestion
func TestShimProcessInnerInstructions_OutOfBoundsStartIndexShouldFail(t *testing.T) {
```
This name is super long but the point is just to clarify that what's being tested here is the "start index" value relative to the length of the inner instructions.